packages/typespec-vscode/src/update-import-on-file-rename.ts

```diff
@@ -0,0 +1,123 @@
+import vscode from "vscode";
+import { State } from "vscode-languageclient";
+import logger from "./log/logger.js";
+import { getAnyExtensionFromPath } from "./path-utils.js";
+import { TspLanguageClient } from "./tsp-language-client.js";
+import { MoveOrRenameAction, SettingName } from "./types.js";
+import { createPromiseWithCancelAndTimeout } from "./utils.js";
+
+export async function updateImportsOnFileMovedOrRenamed(
+  e: vscode.FileRenameEvent,
+  client?: TspLanguageClient,
+): Promise<void> {
+  if (client && client.state === State.Running) {
+    const tspExtensionName = ".tsp";
+    const renames: MoveOrRenameAction[] = [];
+
+    for (const { newUri, oldUri } of e.files) {
+      // Skip if the file is not a TypeScript file
+      if (getAnyExtensionFromPath(newUri.path) !== tspExtensionName) {
+        continue;
+      }
+
+      if (getAnyExtensionFromPath(oldUri.path) !== tspExtensionName) {
+        continue;
+      }
+
+      const setting = vscode.workspace
+        .getConfiguration()
+        .get(SettingName.UpdateImportsOnFileMovedOrRenamed, true);
+      if (!setting) {
+        continue;
+      }
+
+      renames.push({
+        newFilePath: newUri.fsPath,
+        oldFilePath: oldUri.fsPath,
+      });
+
+      vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Window,
+          title: "Checking for update of TSP imports",
+        },
+        async (_progress, token) => {
+          const TIMEOUT = 300000; // set timeout to 5 minutes which should be enough for changes imports
+          await createPromiseWithCancelAndTimeout(flushRenames(client), token, TIMEOUT);
+        },
+      );
+    }
+
+    async function flushRenames(client?: TspLanguageClient): Promise<void> {
+      const tspOpenedFiles = vscode.workspace.textDocuments.filter(
+        (doc) => getAnyExtensionFromPath(doc.fileName) === tspExtensionName,
+      );
+      if (tspOpenedFiles && tspOpenedFiles.length <= 0) {
+        return;
```

Comment:
so if no file is opened, we will do nothing? I think the rename should work no matter whether any file is actually opened in editor, isn't it?